docs/source/tutorials/qmd/07-groups.qmd

```diff
@@ -109,22 +110,25 @@ Next, we create the group for our spline coefficient in a very similar way.
 
 ```{python}
 class SplineCoef(lsl.Group):
-    def __init__(self, name: str, penalty: Array, tau2: lsl.Param) -> None:
+    def __init__(self, name: str, penalty: Array, tau2: lsl.param) -> None:
         penalty_var = lsl.Var(penalty, name=f"{name}_penalty")
 
-        # we save the rank of the penalty matrix here to use it later in our
-        # gibbs kernel
-        rank = lsl.Data(np.linalg.matrix_rank(penalty), _name=f"{name}_rank")
+        evals = jax.numpy.linalg.eigvalsh(penalty)
+        rank = lsl.Data(jnp.sum(evals > 0.0), _name=f"{name}_rank")
+        _log_pdet = jnp.log(jnp.where(evals > 0.0, evals, 1.0)).sum()
+        log_pdet = lsl.Data(_log_pdet, _name=f"{name}_log_pdet")
 
         prior = lsl.Dist(
             MultivariateNormalDegenerate.from_penalty,
             loc=0.0,
             var=tau2,
             pen=penalty_var,
+            rank=rank,
+            log_pdet=log_pdet
```

Comment:
Maybe we want to make the functions to compute the rank an the log pseudo determinant public?

Reply:
Fair point. I thought about that too, but the problem that I see is: from a code-organization point of view, I think it could be confusing and hard to predict to export a `rank` and `log_pdet` function from a module that provides a distribution class. And probably we do not really want to get in the business of providing specific linear algebra functions publicly at the moment. So my approach currently would be to not touch that.

Reply:
Very true. I mean, we also have the utility functions for the IWLS, maybe, in the future we could have a separate module for this kind of functions